src/type/mont_symkey.js

```diff
@@ -0,0 +1,64 @@
+// OpenPGP.js - An OpenPGP implementation in javascript
+// Copyright (C) 2015-2016 Decentral
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+/**
+ * Encoded symmetric key for x25519
+ * The payload format varies for v3 and v6 PKESK:
+ * the former includes an algorithm byte preceeding the encrypted session key.
+ *
+ * @module type/mont_symkey
+ */
+
+import util from '../util';
+
+class MontgomerySymmetricKey {
```

Comment:
I think this class name is a bit confusing if you don't already know what is meant; how about
```suggestion
class AlgoAndSessionKey {
```
(even though the algorithm isn't always there of course, but that still seems clearer to me)

Reply:
This is named based on the ECDHSymKey class; since it applies to Montgomery curves specifically, I'd leave the explicit reference to the publicKeyAlgo, because the two classes are equivalent if no algorithm is present. Maybe `ECDHMontgomery` is clearer?

Reply:
But the difference has nothing to do with the fact that they're Montgomery curves, and furthermore they're never referred to as such in the OpenPGP standard, so someone familiar only with that would be confused I think.
You could call it `X25519X448SymKey` or something like that? Bit ugly but more explicit.